SmartDeviceLink/SDLArtwork.m

```diff
@@ -36,21 +35,70 @@ + (instancetype)persistentArtworkWithImage:(UIImage *)image name:(NSString *)nam
 #pragma mark Private Lifecycle
 
 - (instancetype)initWithImage:(UIImage *)image name:(NSString *)name persistent:(BOOL)persistent asImageFormat:(SDLArtworkImageFormat)imageFormat {
-    NSData *imageData = nil;
-    NSString *fileExtension = nil;
+    return [super initWithData:[self sdl_dataForUIImage:image imageFormat:imageFormat] name:name fileExtension:[self sdl_fileExtensionForImageFormat:imageFormat] persistent:persistent];
+}
 
+- (instancetype)initWithImage:(UIImage *)image persistent:(BOOL)persistent asImageFormat:(SDLArtworkImageFormat)imageFormat {
+    NSData *imageData = [self sdl_dataForUIImage:image imageFormat:imageFormat];
+    NSString *imageName = [self sdl_md5HashFromNSData:imageData];
+    return [super initWithData:[self sdl_dataForUIImage:image imageFormat:imageFormat] name:(imageName != nil ? imageName : @"") fileExtension:[self sdl_fileExtensionForImageFormat:imageFormat] persistent:persistent];
```

Comment:
We may want to return nil if the `imageName` doesn't work, or throw an exception. Having a blank name is going to cause problems.

Reply:
The `imageName` is only `nil` if the `data` is `nil`. Since the `data` is nil, the upload request is going to fail anyways. Developers should be checking if the `data` is `nil` before attempting to send. 

I don't think we should add an assert statement because if developers are creating images dynamically, they aren't going to want the app to crash in the rare instance that the UIImage conversion to a PNG or JPG fails.

Reply:
Added error handling `nil` or empty `data`